vulnerabilities/scraper/ruby.py

```diff
@@ -0,0 +1,104 @@
+import os
+import urllib.request
+import saneyaml
+
+from shutil import rmtree
+from urllib.error import HTTPError
+from zipfile import ZipFile
+from itertools import chain
+from dephell_specifier import RangeSpecifier
+
+RUBYCVE_LINK = 'https://github.com/rubysec/ruby-advisory-db/archive/master.zip'
+DOWNLOAD_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+def get_rubycve_db():
+    pathToZip, _ = urllib.request.urlretrieve(
+        RUBYCVE_LINK, os.path.join(
+            DOWNLOAD_PATH, 'ruby.zip'))
+    ZipFile(pathToZip).extractall(DOWNLOAD_PATH)
+    os.remove(pathToZip)
+
+
+def path_of_yaml_of_all_packages():
+    gemPath = os.path.join(DOWNLOAD_PATH, 'ruby-advisory-db-master', 'gems')
+    rubiesPath = os.path.join(
+        DOWNLOAD_PATH,
+        'ruby-advisory-db-master',
+        'rubies')
+    for (
+            packagePath,
+            _,
+            yamlNames) in chain(
+            os.walk(gemPath),
+            os.walk(rubiesPath)):
+        for yamlName in yamlNames:
+            yield os.path.join(packagePath, yamlName)
+
+
+def get_all_versions_of_package(package_name):
+    url_to_load = 'https://rubygems.org/api/v1/versions/' + package_name + '.yaml'
+    try:
+        page = urllib.request.urlopen(url_to_load)
+        package_history = saneyaml.load(page)
+    except HTTPError:
+        return []
+    for version in package_history:
+        yield version['number']
+
+
+def get_patched_range(spec_list):
+    if spec_list:
+        def remove_space(string): return string.replace(' ', '')
+        spec_list = list(map(remove_space, spec_list))
+        for spec in spec_list:
+            if 'rc' in spec:
+                continue
+            yield RangeSpecifier(spec)
+    else:
+        return None
+
+
+def import_vulnerabilities():
+    get_rubycve_db()
+    ids = set()
+    vulnerability_to_package_map = []
```

Comment:
It's a list of dictionaries, each dictionary mapping a vulnerability to affected package. Any better names you can think of ? Naming is tough

Reply:
I have changed it `vulnerability_package_dicts` for now

Reply:
Yes, naming is hard. In general I think it's preferable to avoid describing the data structure in the variable name. But let's not get hung up on this here and stick with the current name.